ci-operator/REHEARSALS.md

```diff
@@ -0,0 +1,39 @@
+CI job rehearsals
+=================
+
+**This feature is not fully deployed to production yet**
+
+Pull requests to this repository that contain changes affecting Prow job setup
+trigger a so-called "rehearsal". Jobs that would be affected by a PR are
+executed as if run against a target component repository after job changes
+would be merged. This provides job config authors early feedback about how job
+config changes impact CI setup for a given repo.
+
+## How rehearsal works
+
+All pull requests trigger a `ci/prow/pj-rehearse` check, which detects the jobs
+affected by the proposed change. For each job selected for rehearsal (for
+various reasons, we are not able to rehearse all jobs), new Prowjob is
+dynamically created, reporting results via a GitHub context named with the
+`ci/rehearse/$org/$repo/$test` pattern.
+
+## Which jobs are rehearsed
+
+At the moment, we are not able to rehearse all affected jobs for various
+reasons. At the moment, we only rehearse `Presubmit` jobs affected by the
+following changes:
+
+1. Newly added presubmit jobs
+2. Presubmit jobs whose `spec:` field was changed
+
+These jobs are then further filtered to exclude jobs where some factor prevents
+us from reliably rehearsing them. Decisions about why a job was excluded from
+rehearsing are logged in the output of the `ci/prow/pj-rehearse` job. Most
+importantly, we are currently *not* rehearsing template-based jobs.
+
+## Future work
```

Comment:
Worth to mention, that in the future we will rehearse job not only for changed `spec`, but for other fields as well.

Reply:
addressed